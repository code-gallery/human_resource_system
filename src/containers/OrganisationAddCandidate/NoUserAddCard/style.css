.NoUserAddCard {
  box-sizing: border-box;
  position: relative;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 0;
  padding: 27px 30px 27px 19px;
  font-family: "Graphik-Regular", sans-serif;
  font-weight: 400;
  -webkit-font-smoothing: antialiased;
  border: 1px solid #E7E5EE;
  background-color: #fff;
}

.NoUserAddCard.is-loading {
  justify-content: center;
}

.NoUserAddCard__txt-area {
  display: block;
  width: 100%;
}

.NoUserAddCard__msg {
  margin-top: 0;
  margin-bottom: 3px;
  font-size: 15px;
  line-height: 18px;
  color: #212224;
}

.NoUserAddCard__txt {
  margin: 0;
  font-size: 13px;
  line-height: 16px;
  color: #7F8DAA;
}

.NoUserAddCard__right {
  box-sizing: border-box;
  position: relative;
  display: flex;
  justify-content: flex-end;
  width: 100%;
}

.NoUserAddCard__right--Added {
  width: 150px;
}

/** @NOTE: Repeated style. Will be extracted to component. */
.NoUserAddCard__btn {
  all: initial;
  box-sizing: border-box;
  display: block;
  margin-left: 16px;
  font-family: "Graphik-Medium";
  font-weight: 600;
  -webkit-font-smoothing: antialiased;
  font-size: 11px;
  line-height: 13px;
  text-transform: uppercase;
  letter-spacing: 0.075em;
  color: #498DF0;
  cursor: pointer;
}

.NoUserAddCard__btn--green {
  color: #72d371;
}

.NoUserAddCard__input {
  all: initial;
  box-sizing: border-box;
  display: block;
  width: 100%;
  padding: 10px 14px;
  font-family: "Graphik-Regular", sans-serif;
  font-weight: 400;
  -webkit-font-smoothing: antialiased;
  font-size: 15px;
  line-height: 24px;
  border-bottom: 1px solid #E7E5EE;
  border-radius: 3px;
  color: #44484C;
}

.NoUserAddCard__input::placeholder {
  font-family: "Graphik-Regular", sans-serif;
  font-weight: 400;
  -webkit-font-smoothing: antialiased;
  color: #7F8DAA;
}

.NoUserAddCard__input:focus {
  border-color: #7F8DAA;
}
