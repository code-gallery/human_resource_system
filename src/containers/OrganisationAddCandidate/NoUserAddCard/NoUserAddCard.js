import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import Loader from 'components/Loader'
import './style.css'

/**
 <NotAdded />
 */
class NotAdded extends Component {
  handleSubmit = (evt) => {
    evt.preventDefault()
    this.props.onAdd(this.input.value)
    this.input.value = ''
  }

  handleRef = (elem) => {
    this.input = elem
  }

  render() {
    const { className } = this.props

    return (
      <section className={`NoUserAddCard ${className}`}>
        <div className="NoUserAddCard__txt-area">
          <p className="NoUserAddCard__msg">No results</p>
          <p className="NoUserAddCard__txt">
            Add candidate&apos;s email to create a request
          </p>
        </div>

        <form className="NoUserAddCard__right" onSubmit={this.handleSubmit}>
          <input
            className="NoUserAddCard__input"
            type="email"
            placeholder="example@domain.com"
            ref={this.handleRef}
            required
          />
          <button className="NoUserAddCard__btn" type="submit">
            ADD
          </button>
        </form>
      </section>
    )
  }
}

NotAdded.propTypes = {
  onAdd: PropTypes.func.isRequired,
  className: PropTypes.string
}

NotAdded.defaultProps = {
  className: ''
}

/**
 <Added />
 */
const Added = ({ candidateEmail, newRequestLink, className }) => (
  <section className={`NoUserAddCard ${className}`}>
    <div className="NoUserAddCard__txt-area">
      <p className="NoUserAddCard__msg">
        {candidateEmail} was added to your candidate list.
      </p>
      <p className="NoUserAddCard__txt">
        They&apos;ll be sent an invite when you create a request.
      </p>
    </div>

    <div className="NoUserAddCard__right NoUserAddCard__right--Added">
      <Link
        className="NoUserAddCard__btn NoUserAddCard__btn--green"
        to={newRequestLink}
      >
        Create Request
      </Link>
    </div>
  </section>
)

Added.propTypes = {
  candidateEmail: PropTypes.string.isRequired,
  newRequestLink: PropTypes.string.isRequired,
  className: PropTypes.string
}

Added.defaultProps = {
  className: ''
}

/**
 <NoUserAddCard />
 */
const NoUserAddCard = ({
  loading,
  added,
  onAdd,
  candidateEmail,
  newRequestLink,
  className
}) => {
  if (loading) {
    return (
      <section className="NoUserAddCard is-loading">
        <Loader size={50} color="#72d371" />
      </section>
    )
  }

  return !added
    ? <NotAdded className={className} onAdd={onAdd} />
    : (
      <Added
        className={className}
        candidateEmail={candidateEmail}
        newRequestLink={newRequestLink}
      />
    )
}

NoUserAddCard.propTypes = {
  loading: PropTypes.bool.isRequired,
  added: PropTypes.bool.isRequired,
  onAdd: PropTypes.func,
  candidateEmail: PropTypes.string,
  newRequestLink: PropTypes.string,
  className: PropTypes.string
}

NoUserAddCard.defaultProps = {
  className: ''
}

export default NoUserAddCard
