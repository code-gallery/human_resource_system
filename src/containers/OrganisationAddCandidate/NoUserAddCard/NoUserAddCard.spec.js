import React from 'react'
import renderer from 'react-test-renderer'
import NoUserAddCard from '.'

describe('<NoUserAddCard />', () => {
  const props = {
    loading: false,
    added: false,
    onAdd: jest.fn()
  }

  describe('Rendering', () => {
    it('renders without crashing', () => {
      const tree = renderer.create(
        <NoUserAddCard {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
