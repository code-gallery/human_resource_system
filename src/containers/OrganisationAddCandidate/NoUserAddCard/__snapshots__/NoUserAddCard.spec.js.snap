// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoUserAddCard /> Rendering renders without crashing 1`] = `
<section
  className="NoUserAddCard "
>
  <div
    className="NoUserAddCard__txt-area"
  >
    <p
      className="NoUserAddCard__msg"
    >
      No results
    </p>
    <p
      className="NoUserAddCard__txt"
    >
      Add candidate's email to create a request
    </p>
  </div>
  <form
    className="NoUserAddCard__right"
    onSubmit={[Function]}
  >
    <input
      className="NoUserAddCard__input"
      placeholder="example@domain.com"
      required={true}
      type="email"
    />
    <button
      className="NoUserAddCard__btn"
      type="submit"
    >
      ADD
    </button>
  </form>
</section>
`;
